Tensile/KernelWriter.py

```diff
@@ -119,74 +114,174 @@ def makeSchedule(self, kernel, tensorParametersA, tensorParametersB, localWriteE
+            if self.numReadPerVectorA != 1 or self.numReadPerVectorB !=1:
+              # fp16 or bf16, we read 1 element to vgprBuffer the other element to tempVgpr.
+              # since each iteration shares same tempVgpr, only read-to-vgprBuffer can
+              # be scheduled in the front of loop.
+              # localwrite have to start after last read-to-tempVgpr.
+              numHalfReads = (self.numReadPerVectorA//2)*kernel["InnerUnroll"]*kernel["MIWaveTileA"] + (self.numReadPerVectorB//2)*kernel["InnerUnroll"]*kernel["MIWaveTileB"]
+              numMfmaForHalfRead = 1
+              latencyLeft = self.miLatencyLeft
+              for i in range(numHalfReads):
+                latencyLeft -= 2
+                if latencyLeft < 0:
+                  numMfmaForHalfRead += 1
+                  latencyLeft = max(self.miLatencyLeft - 2, 0)
+              lwStartMfmaIndex = numMfmaPerIter * (kernel["LoopIters"] - 1 - self.numItersPLR) + numMfmaForHalfRead
+            else:
+              numReads = (self.numReadsPerIterA + self.numReadsPerIterB) * (kernel["LoopIters"]//(kernel["LocalReadVectorWidth"]//kernel["MIInputPerThread"]) - self.numItersPLR)
+              numMfmaForCurrentLoopLR = 1
+              latencyLeft = self.miLatencyLeft
+              for i in range(numReads):
+                latencyLeft -= tensorParametersB["localReadInstruction"].IssueLatency*2
+                if latencyLeft < 0:
+                  numMfmaForCurrentLoopLR += 1
+                  latencyLeft = max(self.miLatencyLeft - tensorParametersB["localReadInstruction"].IssueLatency*2,0)
+              if kernel["MIWaveTileA"] * kernel["MIWaveTileB"] > 1:
+                numMfmaForCurrentLoopLR += 1
+              lwStartMfmaIndex = numMfmaForCurrentLoopLR
+          else:
+            lwStartMfmaIndex = numMfmaPerIter * (kernel["LoopIters"] - 1 - self.numItersPLR) + self.numMfmaForLR
+          # to calculate number of mfma we need to wait before data arrive from lds to vgpr.
+          # latency: 40 quad-cycle for 4 word, 20 quad-cycle for 2 word, 10 quad-cycle for 1 word / half word
+          latencyForLR = roundUp(tensorParametersB["localReadInstruction"].blockWidth)*10
+          latencyForLR -= max(latencyLeft,0) # remaining latency in mfma
+          while latencyForLR > 0:
+            latencyForLR -= self.miLatency
+            lwStartMfmaIndex += 1
+        #########
+        # Get LocalWritePerMfma
+        #########
+        if lwStartMfmaIndex > self.lwEndMfmaIndex:
+          lwStartMfmaIndex = self.lwEndMfmaIndex
+        numMfmaCanSched = self.lwEndMfmaIndex - lwStartMfmaIndex + 1
+        numLoadsA = kernel["DepthU"]*kernel["MacroTileA"]//kernel["GlobalLoadVectorWidthA"]//kernel["NumThreads"]
+        numLoadsB = kernel["DepthU"]*kernel["MacroTileB"]//kernel["GlobalLoadVectorWidthB"]//kernel["NumThreads"]
+        writesToSched = (numLoadsA + numLoadsB - 1) * 100
+        temp1 = 0
+        temp2 = 100
```

Comment:
variable names and hardcoded numbers seem a bit of a hack - should probably have better names or comments to document what's happening here